promql/parser/printer.go

```diff
@@ -128,19 +128,24 @@ func (node *BinaryExpr) ShortString() string {
 func (node *BinaryExpr) getMatchingStr() string {
 	matching := ""
 	vm := node.VectorMatching
-	if vm != nil && (len(vm.MatchingLabels) > 0 || vm.On) {
-		vmTag := "ignoring"
-		if vm.On {
-			vmTag = "on"
-		}
-		matching = fmt.Sprintf(" %s (%s)", vmTag, strings.Join(vm.MatchingLabels, ", "))
-
-		if vm.Card == CardManyToOne || vm.Card == CardOneToMany {
-			vmCard := "right"
-			if vm.Card == CardManyToOne {
-				vmCard = "left"
+	if vm != nil {
+		if len(vm.MatchingLabels) > 0 || vm.On {
```

Comment:
was there a reason for removing the &&? The code seems equivalent and the new form implies that there might be an `else` below, though there is not.